Here are two fixes for compiling quickcam module with newer kernels.

In 2.6.27-rc1 :
1) the type field was removed from the video_device structure.
2) the parent of the video device was renamed from dev to parent.

--- qc-usb-0.6.6/qc-driver.c.2.6.27	2008-10-15 21:40:01.000000000 +0200
+++ qc-usb-0.6.6/qc-driver.c	2008-10-15 21:40:01.000000000 +0200
@@ -2528,7 +2528,7 @@
 			if (qcdebug&QC_DEBUGUSER) PDEBUG("VIDIOCGCAP");
 			memset(&b, 0, sizeof(b));
 			strcpy(b.name, "Logitech QuickCam USB");	/* Max 31 characters */
-			b.type      = qc->vdev.type;
+			b.type      = VID_TYPE_CAPTURE | VID_TYPE_SUBCAPTURE;
 			b.channels  = 1;
 			b.audios    = 0;
 			b.maxwidth  = qc->sensor_data.maxwidth;
@@ -3007,7 +3007,6 @@
 
 static struct video_device qc_v4l_template = {
 	name:		"QuickCam USB",
-	type:		VID_TYPE_CAPTURE | VID_TYPE_SUBCAPTURE,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 	hardware:	VID_HARDWARE_QCAM_USB,
 #endif
